app/assets/javascripts/discourse/app/lib/sidebar/admin-sidebar.js

```diff
@@ -405,6 +405,19 @@ export default class AdminSidebarPanel extends BaseCustomSidebarPanel {
       navConfig = navConfig.filterBy("links.length");
     }
 
+    if (!siteSettings.enable_group_directory) {
```

Comment:
Not sure if we necessarily need to gate it on this setting, made a post about it in our internal topic